Kudu.Core/Jobs/AggregrateJobsManagerBase.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Kudu.Contracts.Jobs;
+using Kudu.Contracts.Settings;
+using Kudu.Core.Tracing;
+
+namespace Kudu.Core.Jobs
+{
+    public abstract class AggregrateJobsManagerBase<TJob> where TJob : JobBase, new()
+    {
+        protected JobsManagerBase<TJob> PrimaryJobManager { get; private set; }
+        protected JobsManagerBase<TJob> SecondaryJobManager { get; private set; }
+        protected JobsManagerBase<TJob> WriteJobManager { get; private set; }
+
+        private readonly IDeploymentSettingsManager _settings;
+
+        protected AggregrateJobsManagerBase(Func<JobsManagerBase<TJob>> primaryManagerAction, Func<IEnumerable<string>, JobsManagerBase<TJob>> secondaryManagerAction, IDeploymentSettingsManager settings)
+        {
+            PrimaryJobManager = primaryManagerAction();
+            SecondaryJobManager = secondaryManagerAction(PrimaryJobManager.ListJobs(false).Select(j => j.Name));
+            WriteJobManager = settings.RunFromLocalZip() || settings.RunFromRemoteZip() ? SecondaryJobManager : PrimaryJobManager;
+            _settings = settings;
+        }
+
+        public void DeleteJob(string jobName)
+        {
+            if (_settings.RunFromLocalZip() || _settings.RunFromRemoteZip())
```

Comment:
Suggestion, use a wrapper `RunFromZip()` helper for cases where you need either.

Reply:
done